### PR TITLE
feat(testing): add pre-configuration for vitest

### PR DESCRIPTION
**Pull request by tkint**
Just adding a pre-configuration for vitest in `createTestingPinia`

### CODE REVIEW

File: packages/testing/src/testing.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export interface TestingOptions {
 
   /**
    * Function used to create a spy for actions and `$patch()`. Pre-configured
-   * with `jest.fn()` in jest projects.
+   * with `jest.fn()` in jest projects or `vitest.fn()` in vitest projects
```
Comment (posva) [inline packages/testing/src/testing.ts]:
```suggestion
   * with `jest.fn()` in jest projects or `vitest.fn()` in vitest projects.
```

File: packages/testing/src/testing.ts [hunk 2]
```diff
@@ -64,8 +64,8 @@ export interface TestingPinia extends Pinia {
  * Creates a pinia instance designed for unit tests that **requires mocking**
  * the stores. By default, **all actions are mocked** and therefore not
  * executed. This allows you to unit test your store and components separately.
- * You can change this with the `stubActions` option. If you are using jest,
- * they are replaced with `jest.fn()`, otherwise, you must provide your own
+ * You can change this with the `stubActions` option. If you are using jest or vitest,
+ * they are replaced with basic `fn()`, otherwise, you must provide your own
```
Comment (posva) [inline packages/testing/src/testing.ts]: let's revert this changes, having one example with jest is enough here

File: packages/testing/src/testing.ts [hunk 3]
```diff
@@ -89,7 +89,7 @@ export function createTestingPinia({
 
   plugins.forEach((plugin) => pinia.use(plugin))
 
-  const createSpy = _createSpy || (typeof jest !== 'undefined' && jest.fn)
+  const createSpy = _createSpy || (typeof jest !== 'undefined' && jest.fn) || (typeof vitest !== 'undefined' && vitest.fn)
```
Comment (posva) [inline packages/testing/src/testing.ts:92]: Isn't it 
```suggestion
  const createSpy = _createSpy || (typeof jest !== 'undefined' && jest.fn) || (typeof vi !== 'undefined' && vi.fn)
```

**Review by posva — CHANGES_REQUESTED** [review]

File: packages/testing/src/testing.ts (hunk 3 above)
Comment (tkint) [reply packages/testing/src/testing.ts:92]: Both of them are valid: https://github.com/vitest-dev/vitest/blob/main/packages/vitest/globals.d.ts

**Review by posva — CHANGES_REQUESTED** [review]
Can you fix the tests? (formatting so far)

**Comment by posva** [comment]
Thanks, I merged this locally 44f0762
